CMakeLists.txt

```diff
@@ -1,10 +1,14 @@
-cmake_minimum_required(VERSION 3.20)
+cmake_minimum_required(VERSION 3.29)
```

Comment:
What do we gain by bumping this? We should try to keep tooling versions as low as reasonably possible unless we have a reason to update.